### PR TITLE
Update to Hive 3.0.0

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: src/main/java/org/apache/hive/common/util/TimestampParser.java [hunk 1]
```diff
@@ -1,191 +0,0 @@
-/**
```
Comment (findepi) [inline src/main/java/org/apache/hive/common/util/TimestampParser.java:1]: ❤️

File: src/main/java/org/apache/hive/hcatalog/data/JsonSerDe.java [hunk 2]
```diff
@@ -106,26 +109,23 @@ public void initialize(Configuration conf, Properties tbl)
...
     // Get column names and types
     String columnNameProperty = tbl.getProperty(serdeConstants.LIST_COLUMNS);
     String columnTypeProperty = tbl.getProperty(serdeConstants.LIST_COLUMN_TYPES);
     final String columnNameDelimiter = tbl.containsKey(serdeConstants.COLUMN_NAME_DELIMITER) ? tbl
         .getProperty(serdeConstants.COLUMN_NAME_DELIMITER) : String.valueOf(SerDeUtils.COMMA);
     // all table column names
-    if (columnNameProperty.length() == 0) {
-      columnNames = new ArrayList<String>();
+    if (columnNameProperty.isEmpty()) {
+      columnNames = Collections.emptyList();
```
Comment (findepi) [inline src/main/java/org/apache/hive/hcatalog/data/JsonSerDe.java:121]: any other copied classes need update?

File: pom.xml [hunk 3]
```diff
@@ -43,14 +43,13 @@
 
         <shadeBase>io.prestosql.hive.\$internal</shadeBase>
 
-        <dep.hive.version>2.3.4</dep.hive.version>
+        <dep.hive.version>3.0.0</dep.hive.version>
```
Comment (findepi) [inline pom.xml:46]: I expect that Hive 3.0 is not an interesting version. Everybody who upgrades to Hive 3 will probably upgrade to 3.1.
I am ok with upgrading to 3.0 if
- we later upgrade to 3.1
- or we realize that my expectations do not match real world

Comment (findepi) [reply pom.xml:46]: nvrm; https://github.com/prestosql/presto/pull/198

File: src/main/java/org/apache/hive/common/util/TimestampParser.java (hunk 1 above)
Comment (electrum) [reply src/main/java/org/apache/hive/common/util/TimestampParser.java:1]: 👍 They merged your changes

File: src/main/java/org/apache/hive/hcatalog/data/JsonSerDe.java (hunk 2 above)
Comment (electrum) [reply src/main/java/org/apache/hive/hcatalog/data/JsonSerDe.java:121]: Yes, I verified the others (most are shims and not copied classes)

File: pom.xml (hunk 3 above)
Comment (electrum) [reply pom.xml:46]: I agree, it's not interesting. This is to update the Parquet version since the one in `2.3.4` is broken.

**Review by dain — APPROVED** [review]

**Pull request merged**